Dockerfile

```diff
@@ -17,7 +18,9 @@ RUN apt-get update && \
     curl \
     make=${APT_MAKE_VERSION} \
     gcc=${APT_GCC_VERSION} \
-    git=${APT_GIT_VERSION} \
+    # TODO: figure out git version being used
```

Comment:
I don't know if this tells you what you want, but if you're just trying to version lock git, this is the version that comes with your base image:

```
donovan@jupiter:~$ docker run --rm -it golang:1.21.1-bullseye /bin/bash
Unable to find image 'golang:1.21.1-bullseye' locally

1.21.1-bullseye: Pulling from library/golang
ddf874abf16c: Pull complete 
5c1459d3ab8b: Pull complete 
29ab00e7798a: Pull complete 
9036a471a1c4: Pull complete 
bf598f34e2fc: Pull complete 
af347894f6dc: Pull complete 
Digest: sha256:18f57bdd1d188ade1226471393b5c3ced4639c6c772fd94853ea7abd9eb158f7
Status: Downloaded newer image for golang:1.21.1-bullseye

root@007df2a542c7:/go# 
root@007df2a542c7:/go# dpkg -l git
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version            Architecture Description
+++-==============-==================-============-=============================
======================
ii  git            1:2.30.2-1+deb11u2 amd64        fast, scalable, distributed r
evision control system
```

Reply:
This is super helpful! Thank you for sharing the steps :D